src/FileDownloader.ts

```diff
@@ -107,6 +112,16 @@ export default class FileDownloader implements IFileDownloader {
             }
             throw error;
         }
+        finally {
+            if (shouldUnzip) {
+                try {
+                    await rimrafAsync(tempZipFileDownloadPath);
```

Comment:
Could you please clarify why doing this here, rather than right in `unzipDownloadedFileAsyncFn`? Moving it there would have the advantage that:
- All ZIP things would be together.
- The deletion would also benefit from the exponential retry (as it might fail).